File: app/src/main/java/org/tm/archive/components/settings/app/about/AboutPreferenceFragment.java
package org.tm.archive.components.settings.app.about;

import android.content.Context;
import android.os.Bundle;
import android.os.Handler;
import android.text.Html;
import android.text.method.LinkMovementMethod;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.LinearLayout;
import android.widget.TextView;

import androidx.fragment.app.Fragment;

import org.tm.archive.BuildConfig;
import org.tm.archive.R;
//**TM_SA**//Add this class
public class AboutPreferenceFragment extends Fragment {
  private static final String TAG = AboutPreferenceFragment.class.getSimpleName();

  private TextView fullAppName;
  private TextView version;
  private TextView doNotSell;
  private TextView policy;
  private TextView center;
  private TextView terms;
  private LinearLayout contactUs;
  private LinearLayout issueR;
  protected Handler mHandler = new Handler();
  private final String LOG_NAME = "log.zip";


  @Override
  public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
    View view = inflater.inflate(R.layout.settings_about_custom_layout, container, false);
    //name = (TextView)view.view.findViewById(R.id.Name);
    //location = (TextView)view.view.findViewById(R.id.Location);
    initResources(view);
    return view;
  }

  @Override
  public void onResume() {
    super.onResume();
    //getApplicationPreferencesActivity().getSupportActionBar().setTitle(R.string.ManageProfileFragment_about);
  }


  private void initResources(View view){

    fullAppName = view.findViewById(R.id.full_app_name);
    version = view.findViewById(R.id.version);
    terms = view.findViewById(R.id.terms);
    center = view.findViewById(R.id.privacy_center);


    version.setText("version" + " " + getVersionString(getActivity()));

    setString(fullAppName, R.string.settings_about_title, null);
    setString(terms, R.string.terms_about, null);
    setString(center, R.string.privacy_center_about, null);

  }


  private void setString(TextView view, int strId,String str){
    String text = strId>0 ? getString(strId) : str;
    view.setText(Html.fromHtml(text));
    view.setMovementMethod(LinkMovementMethod.getInstance());
//        stripUnderlines(view);
  }


  public static String getVersionString(Context context){
    String versionName;

    //  versionName = context.getPackageManager().getPackageInfo(context.getPackageName(), 0).versionName;
      versionName = BuildConfig.signal_teleMessage_version;

    return versionName;
  }



}

File: app/src/main/java/org/tm/archive/components/settings/app/about/AboutPreferenceFragment.kt
package org.tm.archive.components.settings.app.about

import android.content.Context
import android.os.Bundle
import android.os.Handler
import android.text.Html
import android.text.method.LinkMovementMethod
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.LinearLayout
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity
import androidx.appcompat.widget.Toolbar
import androidx.fragment.app.Fragment
import org.tm.archive.BuildConfig
import org.tm.archive.R


//**TM_SA**//Add this class
class AboutPreferenceFragment : Fragment() {
    private var fullAppName: TextView? = null
    private var version: TextView? = null
    private val doNotSell: TextView? = null
    private val policy: TextView? = null
    private var center: TextView? = null
    private var terms: TextView? = null
    private var supportCenter: TextView? = null
    private val contactUs: LinearLayout? = null
    private val issueR: LinearLayout? = null
    protected var mHandler = Handler()
    private val LOG_NAME = "log.zip"

  override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        val view = inflater.inflate(R.layout.settings_about_custom_layout, container, false)
        //name = (TextView)view.view.findViewById(R.id.Name);
        //location = (TextView)view.view.findViewById(R.id.Location);

        initializeAppBar(view)
        initResources(view)
        return view
    }

    private fun initializeAppBar(view: View) {
        val primaryToolbar: Toolbar = view.findViewById(R.id.toolbar)
        (activity as AppCompatActivity?)!!.setSupportActionBar(primaryToolbar)
        assert((activity as AppCompatActivity?)!!.supportActionBar != null)
        (activity as AppCompatActivity?)!!.supportActionBar!!.setDisplayHomeAsUpEnabled(true)
        (activity as AppCompatActivity?)!!.supportActionBar!!.setTitle(R.string.EditAboutFragment_about)

      primaryToolbar.setNavigationOnClickListener {
        requireActivity().onBackPressed()
      }

    }

    private fun initResources(view: View) {
        fullAppName = view.findViewById(R.id.full_app_name)
        version = view.findViewById(R.id.version)
        terms = view.findViewById(R.id.terms)
        center = view.findViewById(R.id.privacy_center)
        supportCenter = view.findViewById(R.id.TVSupportCenter)
        version!!.setText(
            "version" + " " + getVersionString(
                activity
            )
        )
        setString(fullAppName, R.string.settings_about_title, null)
        setString(terms, R.string.terms_about, null)
        setString(center, R.string.privacy_center_about, null)
        setString(supportCenter, R.string.support_about, null)
    }

    private fun setString(TextView: TextView?, strId: Int, str: String?) {
        val text = if (strId > 0) getString(strId) else str!!
      TextView!!.text = Html.fromHtml(text)
      TextView.movementMethod = LinkMovementMethod.getInstance()
        //        stripUnderlines(view);
    }

    companion object {
        private val TAG = AboutPreferenceFragment::class.java.simpleName
        fun getVersionString(context: Context?): String {
            val versionName: String

            //  versionName = context.getPackageManager().getPackageInfo(context.getPackageName(), 0).versionName;
            versionName = BuildConfig.VERSION_NAME
            return versionName
        }
    }


}